projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitPdfActivity.java

```diff
@@ -0,0 +1,316 @@
+        fiveLawsuitsStatus = sentMoreThanFiveLawsuits ? moreThanFiveLawsuits: lessThanFiveLawsuits;
+        res = res.replace("&lt;claimCase&gt;", claimCase);
+        res = res.replace("&lt;claimAmount&gt;", claimAmount);
+        res = res.replace("&lt;spamType&gt;", spamType);
+        res = res.replace("&lt;moreThanFiveLawsuits&gt;", fiveLawsuitsStatus);
+        res = res.replace("&lt;undefined&gt;", "");
+        res = res.replace("&lt;receivedSpamDate&gt;", dateFormatter.format(receivedSpamDate));
+        res = res.replace("&lt;currentDate&gt;", dateFormatter.format(currentDate));
+
+        // User details:
+        res = res.replace("&lt;userFullName&gt;", userPrivateName + " " + userLastname);
+        res = res.replace("&lt;userId&gt;", userID);
+        res = res.replace("&lt;userAddress&gt;", userAddress);
+        res = res.replace("&lt;userPhone&gt;", userPhone);
+        res = res.replace("&lt;userFax&gt;", userFax);
+
+        // First company's details:
+        res = res.replace("&lt;companyName&gt;", companyName);
+        res = res.replace("&lt;companyId&gt;", companyId);
+        res = res.replace("&lt;companyAddress&gt;", companyAddress);
+        res = res.replace("&lt;companyPhone&gt;", companyPhone);
+        res = res.replace("&lt;companyFax&gt;", companyFax);
+
+        // Second company's details:
+        res = res.replace("&lt;company2Name&gt;", company2Name);
+        res = res.replace("&lt;company2Id&gt;", company2Id);
+        res = res.replace("&lt;company2Address&gt;", company2Address);
+        res = res.replace("&lt;company2Phone&gt;", company2Phone);
+        res = res.replace("&lt;company2Fax&gt;", company2Fax);
+
+        return res;
+    }
+
+    /**
+     * Responsible for the output lawsuit PDF file name format.
+     * @return File name with ".pdf" extension.
+     */
+    protected String createPdfLawsuitFileName(){
+
+        String fileName = dateAndTimeFormatter.format(new Date()) + ".pdf";
+        return fileName;
+    }
+
+    /**
+     * This is where configurations of folders structure of Lawsuit Activity is been done.
+     * In case where directories don't exists, it creates them.
+     */
+    protected void initDirStructure(){
+
+        // Main LawsuitActivity path: 'app_name/LAWSUIT_MAIN_DIR_NAME/'
+        lawsuitMainPath = Paths.get(Environment.getExternalStorageDirectory().getPath(),getString(R.string.app_name),LAWSUIT_MAIN_DIR_NAME).toString();
+        if (!dirExists(lawsuitMainPath)){
+            File directory = new File(lawsuitMainPath);
+            directory.mkdirs();
+        }
+
+        // Templates dir path: 'lawsuitMainPath/LAWSUIT_TEMPLATE_DIR_NAME/'
+        // TODO: Currently template's file is been stored in user's device storage. Check to see if this should be part of the apk package.
+        lawsuitTemplateDirPath = Paths.get(lawsuitMainPath,LAWSUIT_TEMPLATE_DIR_NAME).toString();
+        if (!dirExists(lawsuitTemplateDirPath)){
```

Comment:
Looks like you could simplify the code by extracting this to a method:
```
        if (!dirExists(lawsuitTemplateDirPath)){
            File directory = new File(lawsuitTemplateDirPath);
            directory.mkdirs();
        }
```

Reply:
Done.